### PR TITLE
Add cronJob and job to RBAC workload tests

### PR DESCRIPTION
**Pull request by aaronRancher**
Adds the last two workload types to RBAC workload tests.

### CODE REVIEW

File: tests/validation/tests/v3_api/test_workload.py [hunk 1]
```diff
@@ -1004,65 +990,59 @@ def test_rbac_wl_daemonset_update(role, remove_resource):
...
-                                                          containers=con,
-                                                          namespaceId=ns.id,
-                                                          daemonSetConfig={})
+        workload = create_workload_by_type(cluster_owner_p_client,
+                                           name, con, ns, config)
         # project read-only can NOT edit existing workload
         with pytest.raises(ApiError) as e:
-            user_token = rbac_get_user_token_by_role(role)
-            project2, ns = create_project_and_ns(user_token, namespace["cluster"],
+            project3, ns = create_project_and_ns(user_token, namespace["cluster"],
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_workload.py:1025]: why do we create a new project here? Should we user the unshared project (project2)?

File: tests/validation/tests/v3_api/test_workload.py [hunk 2]
```diff
@@ -1004,65 +990,59 @@ def test_rbac_wl_daemonset_update(role, remove_resource):
...
-            p_client = get_project_client_for_token(project2, user_token)
+            p_client = get_project_client_for_token(project3, user_token)
             con = [{"name": "test1", "image": "nginx"}]
             p_client.update(workload, containers=con)
         assert e.value.error.status == 403
         assert e.value.error.code == 'Forbidden'
         remove_resource(workload)
         if role == CLUSTER_MEMBER:
-            remove_resource(project)
+            remove_resource(project2)
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_workload.py:1034]: Do not remove the unshared project.

Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_workload.py]: Addressed in changes.

File: tests/validation/tests/v3_api/test_workload.py [hunk 3]
```diff
@@ -861,85 +862,71 @@ def setup_project_by_role(role, remove_resource):
...
 
 
 @if_test_rbac
 @pytest.mark.parametrize("role", RBAC_ROLES)
-def test_rbac_wl_daemonset_list(role, remove_resource):
+@pytest.mark.parametrize("config", WORKLOAD_TYPES)
+def test_rbac_wl_parametrize_list(role, remove_resource, config):
     if role == CLUSTER_MEMBER:
-        p_client, project, ns = setup_daemonset_project_by_role(role)
+        p_client, project, ns = setup_wl_project_by_role(role)
```
[the remark this answers is not in the document]
Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_workload.py:924]: Addressed in changes.

File: tests/validation/tests/v3_api/test_workload.py [hunk 4]
```diff
@@ -861,85 +862,71 @@ def setup_project_by_role(role, remove_resource):
...
+            workload = create_workload_by_type(p_client, name, con, ns, config)
         assert e.value.error.status == 403
         assert e.value.error.code == 'Forbidden'
 
 
 @if_test_rbac
 @pytest.mark.parametrize("role", RBAC_ROLES)
-def test_rbac_wl_daemonset_negative(role, remove_resource):
+@pytest.mark.parametrize("config", WORKLOAD_TYPES)
+def test_rbac_wl_parametrize_negative(role, remove_resource, config):
```
[the remark this answers is not in the document]
Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_workload.py]: Addressed in changes

File: tests/validation/tests/v3_api/test_workload.py [hunk 5]
```diff
@@ -1089,9 +1066,10 @@ def test_rbac_wl_daemonset_delete(role, remove_resource):
 
 @if_test_rbac
 @pytest.mark.parametrize("role", RBAC_ROLES)
-def test_rbac_wl_daemonset_delete_negative(role, remove_resource):
+@pytest.mark.parametrize("config", WORKLOAD_TYPES)
+def test_rbac_wl_parametrize_delete_negative(role, remove_resource, config):
     if role == CLUSTER_OWNER:
-        # cluster owner can edit daemonsets in any project in the cluster
+        # cluster owner can edit workloads in any project in the cluster
```
[the remark this answers is not in the document]
Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_workload.py]: Addressed in changes.

File: tests/validation/tests/v3_api/test_workload.py [hunk 6]
```diff
@@ -1004,65 +990,59 @@ def test_rbac_wl_daemonset_update(role, remove_resource):
...
+        if config == "job":
+            return
+        # cluster owner can edit workloads in any project in the cluster
         user_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
-        p2 = rbac_get_unshared_project()
-        p_client = get_project_client_for_token(project, user_token)
-        p_client2 = get_project_client_for_token(p2, user_token)
-        ns2 = rbac_get_unshared_ns()
+        p_client, project, ns = setup_wl_project_by_role(role)
+        p_client2 = get_project_client_for_token(project, user_token)
```
[the remark this answers is not in the document]
Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_workload.py:1002]: Yes. That was a mistake. Replaced with just one client. Addressed in changes.

File: tests/validation/tests/v3_api/test_workload.py (hunk 1 above)
Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_workload.py:1025]: Changed to use the unshared project. Addressed in changes.

File: tests/validation/tests/v3_api/test_workload.py [hunk 7]
```diff
@@ -1004,65 +990,59 @@ def test_rbac_wl_daemonset_update(role, remove_resource):
...
         name = random_test_name("default")
         con = [{"name": "test1", "image": TEST_IMAGE}]
-        workload = p_client2.create_workload(name=name,
-                                             containers=con,
-                                             namespaceId=ns2.id,
-                                             daemonSetConfig={})
+        workload = create_workload_by_type(p_client2, name, con, ns, config)
         wait_for_wl_to_active(p_client2, workload)
         con = [{"name": "test1", "image": "nginx"}]
         p_client.update(workload, containers=con)
```
[the remark this answers is not in the document]
Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_workload.py:1008]: Addressed in changes.

File: tests/validation/tests/v3_api/test_workload.py [hunk 8]
```diff
@@ -1100,27 +1078,22 @@ def test_rbac_wl_daemonset_delete_negative(role, remove_resource):
         ns = rbac_get_namespace()
         name = random_test_name("default")
         con = [{"name": "test1", "image": TEST_IMAGE}]
-        workload = p_client.create_workload(name=name,
-                                             containers=con,
-                                             namespaceId=ns.id,
-                                             daemonSetConfig={})
+        workload = create_workload_by_type(p_client, name, con, ns, config)
         wait_for_wl_to_active(p_client, workload)
         p_client2.delete(workload)
```
[the remark this answers is not in the document]
Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_workload.py:1081]: Changed to delete from same project. Addressed in changes.

**Comment by aaronRancher** [comment]
Closed in favor of https://github.com/rancher/rancher/pull/27741
Needed to rewrite changes over https://github.com/rancher/rancher/pull/27407

**Pull request closed without merging**